File: src/Portfolio.Domain/ValueObjects/ProcessedTransaction.cs
using Portfolio.Domain.Entities;

namespace Portfolio.Domain.ValueObjects;

public class ProcessedTransaction
{
    public Transaction Transaction { get; private set; }
    public decimal? ProfitLoss { get; set; }
    public decimal? TotalLossAmount { get; set; }
    public bool IsLossDisallowed { get; set; }
    public int? DisallowedByTransactionId { get; set; }
    public List<int> DisallowsPreviousLosses { get; private set; } = [];
    public string? Error { get; set; }

    // Constructor to ensure consistency
    public ProcessedTransaction(Transaction transaction)
    {
        Transaction = transaction ?? throw new ArgumentNullException(nameof(transaction));
    }
}

File: src/Portfolio.Client.Legacy/src/app/core/models/processed-transaction.ts
import { Transaction } from './transaction';

export interface ProcessedTransaction {
    transaction: Transaction;
    profitLoss?: number;
    totalLossAmount?: number;
    isLossDisallowed?: boolean;
    disallowedByTransactionId?: number;
    disallowsPreviousLosses?: number[];
}
